src/lib/ws/helper/probe-ip-limit.ts

```diff
@@ -1,19 +1,62 @@
-import { fetchProbes } from '../server.js';
+import _ from 'lodash';
+import config from 'config';
+import type { fetchProbes as serverFetchProbes, fetchRawSockets as serverFetchRawSockets } from '../server.js';
 import { scopedLogger } from '../../logger.js';
 import { ProbeError } from '../../probe-error.js';
 
+const numberOfProcesses = config.get<number>('server.processes');
+
 const logger = scopedLogger('ws:limit');
 
-export const verifyIpLimit = async (ip: string, socketId: string): Promise<void> => {
-	if (process.env['FAKE_PROBE_IP'] || process.env['TEST_MODE'] === 'unit') {
-		return;
+export class ProbeIpLimit {
+	private timer: NodeJS.Timeout | undefined;
+
+	constructor (
+		private readonly fetchProbes: typeof serverFetchProbes,
+		private readonly fetchRawSockets: typeof serverFetchRawSockets,
+	) {}
+
+	scheduleSync () {
+		clearTimeout(this.timer);
+
+		this.timer = setTimeout(() => {
+			this.syncIpLimit()
+				.finally(() => this.scheduleSync())
+				.catch(error => logger.error(error));
+		}, 60_000 * 2 * Math.random() * numberOfProcesses).unref();
 	}
 
-	const probes = await fetchProbes({ allowStale: false });
-	const previousSocket = probes.find(p => p.ipAddress === ip && p.client !== socketId);
+	async syncIpLimit () {
+		const sockets = await this.fetchRawSockets();
```

Comment:
How about we use `fetchProbes()` here (even `allowStale: true` should be fine I think), and only call `fetchRawSockets()` if we need the socket instance (to disconnect it)?